### ISSUE TITLE
Wrong issue links in plugin changelogs

### ISSUE THREAD

**Issue by millette** [gatsbyjs/gatsby#12755]
## Description

Links to issues appear wrong in many package changelogs. Gatsby's own is ok:
<https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/CHANGELOG.md>

But not so for the plugins, such as:
<https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-image/CHANGELOG.md>

Instead of markdown like:
```markdown
[#12085](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-feed/issues/12085)
```

The manual link could be removed in favor of a simple:
```markdown
#12085
```

### Expected result

Should link issues to right (top-level) URLs.

For example: <https://github.com/gatsbyjs/gatsby/issues/10278>

### Actual result

Links to URLs like <https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-image/issues/10278>

The string `tree/master/packages/gatsby-image/` shouldn't be there.

**Comment by pieh** [comment]
It does seem like default preset for generating our changelogs doesn't handle this correctly. My best guess is that it works for `gatsby` package because in package.json there we point to root of repo:
https://github.com/gatsbyjs/gatsby/blob/8b4d5deb1b766806ec959c061b555ff4cfa08fa4/packages/gatsby/package.json#L171-L174

And in `gatsby-image` we point to directory:
https://github.com/gatsbyjs/gatsby/blob/8b4d5deb1b766806ec959c061b555ff4cfa08fa4/packages/gatsby-image/package.json#L29

It's probably time to fork changelog preset we use and do some customization

**Comment by millette** [comment]
I searched a few times but have no idea where could help with a fix. Any pointers would be appreciated.

**Comment by wardpeet** [comment]
thanks for checking this out @millette 

@pieh will we put the changelog fork in gatsby monorepo or make a separate one?

**Comment by DSchau** [comment]
Bit of a doozy here. As @pieh notes, this works for the gatsby package because the repository URL is the base from which we construct our issues. For all others, the base is _wrong_ because that's not actually where issues are filed in a monorepo.

There are (minimally) two solutions here:

1. PR conventional-changelog to tweak the Handlebars template to account for this
    - Kind of a rabbit hole, but conventional-changelog-core is probably the best place to tweak this
    - See the start of this logic [here](https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-writer/index.js#L23), where a flag is set if the repository should be included, and then the associated Handlebars templates for how it's implemented
1. Use the new `directory` field within the repository object, e.g. see #11145
    - This seems the lowest effort solution, but:
    - Not sure how well supported this is with npmjs.com _and_ other functionality, e.g. `npm open`

I'd propose to solve this we investigate Option #2 more, since it seems a little lower effort (yay!) and seems to be a better solution going forward.

**Comment by moonmeister** [comment]
closed by #15477. Existing links will be fixed in #15630
